lib/app/features/nostr/providers/nostr_cache.dart

```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'nostr_cache.g.dart';
+
+mixin CacheableEvent {
+  String get cacheKey;
+
+  Type get cacheType;
+}
+
+@Riverpod(keepAlive: true)
+class NostrCache extends _$NostrCache {
+  @override
+  Map<Type, Map<String, CacheableEvent>> build() {
+    return {};
+  }
+
+  void cache(CacheableEvent event) {
+    state = {
+      ...state,
+      event.cacheType: {...(state[event.cacheType] ?? {}), event.cacheKey: event},
+    };
+  }
+}
+
+AlwaysAliveProviderListenable<T?> nostrCache<T>(String key) {
```

Comment:
This approach can lead to unexpected behavior and potential memory leaks (https://riverpod.dev/docs/essentials/do_dont)

Reply:
looks good for me